src/Microsoft.ML.Transforms/WhiteningTransform.cs

```diff
@@ -383,249 +507,339 @@ private void TrainModels(Float[][] columnData, int[] rowCounts, IChannel ch)
                         ccol, ccol, ccol, 1, u, ccol, uScaled, ccol, 0, _models[iinfo], ccol);
 
                     if (ex.SaveInv)
                     {
-                        InvModels[iinfo] = new Float[u.Length];
+                        _invModels[iinfo] = new float[u.Length];
                         Mkl.Gemm(Layout, Mkl.Transpose.NoTrans, Mkl.Transpose.NoTrans,
-                            ccol, ccol, ccol, 1, u, ccol, uInvScaled, ccol, 0, InvModels[iinfo], ccol);
+                            ccol, ccol, ccol, 1, u, ccol, uInvScaled, ccol, 0, _invModels[iinfo], ccol);
                     }
                 }
                 else
                     ch.Assert(false);
             }
         }
 
-        private WhiteningTransform(IHost host, ModelLoadContext ctx, IDataView input)
-            : base(host, ctx, input, TestColumn)
-        {
-            Host.AssertValue(ctx);
-
-            // *** Binary format ***
-            // <prefix handled in static Create method>
-            // <base>
-            // foreach added column
-            //   ColInfoEx
-            // foreach model
-            //   whitening matrix
-            //   recovery matrix
-
-            Host.AssertNonEmpty(Infos);
-            _exes = new ColInfoEx[Infos.Length];
-            for (int i = 0; i < _exes.Length; i++)
-                _exes[i] = new ColInfoEx(ctx, Infos[i]);
-
-            _models = new Float[Infos.Length][];
-            InvModels = new Float[Infos.Length][];
-            for (int i = 0; i < Infos.Length; i++)
-            {
-                _models[i] = ctx.Reader.ReadFloatArray();
-                ValidateModel(Host, _models[i], Infos[i].TypeSrc);
-                if (_exes[i].SaveInv)
-                {
-                    InvModels[i] = ctx.Reader.ReadFloatArray();
-                    ValidateModel(Host, InvModels[i], Infos[i].TypeSrc);
-                }
-            }
-            Metadata.Seal();
-        }
-
-        public static WhiteningTransform Create(IHostEnvironment env, ModelLoadContext ctx, IDataView input)
-        {
-            Contracts.CheckValue(env, nameof(env));
-            var h = env.Register(RegistrationName);
-            h.CheckValue(ctx, nameof(ctx));
-            h.CheckValue(input, nameof(input));
-            ctx.CheckAtModel(GetVersionInfo());
-
-            // *** Binary format ***
-            // int: sizeof(Float)
-            // <remainder handled in ctors>
-            int cbFloat = ctx.Reader.ReadInt32();
-            h.CheckDecode(cbFloat == sizeof(Float));
-            return h.Apply("Loading Model", ch => new WhiteningTransform(h, ctx, input));
-        }
-
         public override void Save(ModelSaveContext ctx)
         {
             Host.CheckValue(ctx, nameof(ctx));
             ctx.CheckAtModel();
             ctx.SetVersionInfo(GetVersionInfo());
 
             // *** Binary format ***
-            // int: sizeof(Float)
+            // int: sizeof(float)
```

Comment:
Do we still save out this?

Reply:
I think it simply saves an int, which is the same size as a float, but I'll change that to make it more clear.

Reply:
It does - but I dont see any code that writes that out, so I think you can just remove the comment.

---
In reply to: [229557498](https://github.com/dotnet/machinelearning/pull/1452#discussion_r229557498) [](ancestors = 229557498)

Reply:
It's in SaveColumns I think it starts by saying how many column pairs there are and then saves them.

Reply:
I think that comment refers to older code and not the # of columns. DropColumns did the same:
            // *** Binary format ***
            // int: sizeof(Float)
            // bindings
            int cbFloat = ctx.Reader.ReadInt32();

Either way, it looks like the code prior to your changes did not save out this float - that why I dont the comment is still valid.


---
In reply to: [229559693](https://github.com/dotnet/machinelearning/pull/1452#discussion_r229559693) [](ancestors = 229559693)